File: 2358-number-of-ways-to-split-array/2358-number-of-ways-to-split-array.java
class Solution {
    public int waysToSplitArray(int[] nums) {
        long sum = nums[0];
        for(int i=1;i<nums.length;i++)
        {
            sum+=nums[i];
        }
        int count = 0;
        long l_sum=0;
        for(int i=0;i<nums.length-1;i++)
        {
            l_sum+=nums[i];
            sum-=nums[i];
            if(l_sum>=sum)
                count++;
        }
        return count;
    }
}

File: 2358-number-of-ways-to-split-array/2358-number-of-ways-to-split-array.py
class Solution:
    def waysToSplitArray(self, nums: List[int]) -> int:
        add=sum(nums)
        count, result=0, 0
        for i in range(0,len(nums)-1):
            result+=nums[i]
            add-=nums[i]
            if result>=add:
                count+=1
        return count

        